compensations/src/test/java/org/jboss/narayana/compensations/functional/compensationScoped/MyCompensationHandler.java

```diff
@@ -43,6 +43,8 @@ public void compensate() {
 
         if (myData != null && myData.getData() != null && myData.getData().equals(expectedData)) {
             dataAvailable = true;
+        } else {
+            dataAvailable = false;
```

Comment:
This I have added first thing, to prove that this fails.
The issue was, if there were two instances of this handler, second instance would have myData as null, because of the bug. And because of that it wouldn't change dataAvailability, which was set to true by the first handler instance. So the test would always pass even with the bug